dkc/core/rest/folder.py

```diff
@@ -43,14 +42,31 @@ class Meta:
             'public',
             'access',
         ]
+        # ModelSerializer cannot auto-generate validators for model-level constraints
+        validators = [
+            serializers.UniqueTogetherValidator(
+                queryset=Folder.objects.all(),
+                fields=['parent', 'name'],
+                message='Folder names must be unique among siblings.',
+            ),
+            # This could also be implemented as a UniqueValidator on 'name',
+            # but its easier to not explicitly redefine the whole serializer field
+            serializers.UniqueTogetherValidator(
+                queryset=Folder.objects.filter(parent=None),
+                fields=['name'],
+                message='Root folders must have a unique name.',
+            ),
+            # folder_max_depth and unique_root_folder_per_tree are internal sanity constraints,
```

Comment:
Actually, `folder_max_depth` is something we might want to check at the REST layer (rather than allowing the database to do it). I'm not sure how to do this, so I'll probably file a bug for a future fix if nobody else has suggestions.